lib/ui/platform_dispatcher.dart

```diff
@@ -308,6 +308,63 @@ class PlatformDispatcher {
     _invoke(onMetricsChanged, _onMetricsChangedZone);
   }
 
+  /// A callback invoked when a given platform focus change affects flutter.
```

Comment:
This is truly nit-picky, but I think these docs could be improved a little bit with some more bread crumbs and general expectations. Here is my stab at it. Feel free to completely ignore it or incorporate the bits you like into the doc:

A callback invoked whenever the platform changes which [FlutterView] has input focus.

When the platform moves input focus from one [FlutterView] to another this callback is invoked indicating the new view that has focus and the direction in which focus was received. For example, if focus is moved to the [FlutterView] with ID 2 in the forward direction (e.g. because the user pressed the tab key to move the focus there), the callback receives the following [ViewFocusEvent]: `ViewFocusEvent(2, ViewFocusState.forwardFocused)`. Typically, receivers of this event respond by moving the input focus to the first focusable element inside the [FlutterView] with ID 2. If a view receives focus in the backwards direction (e.g. because the user pressed shift+tab), typically the last focusable element inside that view is focused.

The platform can also move input focus to a non-[FlutterView] element. In that case, this callback is invoked with  [ViewFocusState.unfocus]. Receivers typically respond to this event by removing all focus indications from the app. 

Apps can also programmatically request to move the focus to a desired [FlutterView] by calling [requestViewFocusChange].

The callback is invoked in the same zone in which the callback was set.

See also:
  * [requestViewFocusChange] to programmatically instruct the platform to move focus to a different [FlutterView].
  * [ViewFocusState] for a list of allowed focus transitions.
  * [ViewFocusEvent], which is the event object provided to the callback.

Reply:
Love this doccomment, what a pro!

Reply:
Done